lib/upload.ts

```diff
@@ -985,6 +1014,133 @@ export class BaseUpload {
   _sendRequest(req: HttpRequest, body?: SliceType): Promise<HttpResponse> {
     return sendRequest(req, body, this.options)
   }
+
+  /**
+   * Apply default stall detection options
+   */
+  private _getStallDetectionDefaults(
+    options?: Partial<StallDetectionOptions>
+  ): StallDetectionOptions {
+    return {
+      enabled: options?.enabled ?? true,
+      stallTimeout: options?.stallTimeout ?? 30000,
+      checkInterval: options?.checkInterval ?? 5000,
+      minimumBytesPerSecond: options?.minimumBytesPerSecond ?? 1
```

Comment:
Default option values should be set in `defaultOptions`, so they are all in one place and also properly exposed to users. That being said, I'm not sure if we want to enable this by default in the next release already as we still have to gain experience with this approach and tune it properly. False positive (incorrectly reported stalls) can cause major disruption for tus-js-client users.

Reply:
Would you consider enabling it via an experimental flag or something - or maybe doing a pre-release?

Reply:
We have some good tooling for testing it, we have created a containerised satellite network emulator proxy.

Once the PR starts to solidify I will be putting it through its paces with that through varying network conditions.

I've got a test matrix I can run over it as part of a nightly job.

Reply:
> Would you consider enabling it via an experimental flag or something

Not sure I understand. I'm looking forward to landing this feature in the next releases, but I would just not enable it default for now. If user want to utilize the stall detection, they can always enable this feature using the options.

> Once the PR starts to solidify I will be putting it through its paces with that through varying network conditions.
> I wonder if you'd also be interested in using this tooling?
> I'll open a discussion internally about open sourcing it today.

Absolutely! That sounds fantastic!